students/arun_nalla/session04/trigrams.py

```diff
@@ -0,0 +1,50 @@
+﻿#! usr/bin/env python 3
+#words = 'I wish I may I wish I might'.split()
+def trigram_dict (words):
+    trilist = {}
+    for i in range(len(words)-2):
+        a = tuple(words[i:i+2])
+        b = words[i+2]
+        if a not in trilist:
+            trilist[a] = [b]
+        else:
+            trilist[a].append(b)
```

Comment:
Minor: Lines 8-11 can be reduced to `trilist.setdefault(a, []).append(b)`. Or even better is to use `defaultdict` and get rid of `.setdefault(...)`.